applications/plugins/BeamPlastic/forceField/BeamPlasticFEMForceField.h

```diff
@@ -0,0 +1,658 @@
+    auto vonMisesGradient(const VoigtTensor2 &stressTensor) -> VoigtTensor2;
+    /// Computes the Von Mises yield function hessian (in matrix notation) at a given stress tensor (in Voigt notation)
+    auto vonMisesHessian(const VoigtTensor2 &stressTensor, const double yieldStress) -> VectTensor4;
+
+    //----- Alternative expressions of the above functions with vector notations -----//
+    /// Computes the equivalent stress from a tensor in vector notation
+    double vectEquivalentStress(const VectTensor2 &stressTensor);
+    /// Evaluates the Von Mises yield function for given stress tensor (in vector notation) and yield stress
+    double vectVonMisesYield(const VectTensor2 &stressTensor, const double yieldStress);
+    /// Computes the Von Mises yield function gradient (in vector notation) at a given stress tensor (in vector notation)
+    VectTensor2 vectVonMisesGradient(const VectTensor2 &stressTensor);
+
+    //----- Alternative functions using the deviatoric stress expression -----//
+    // TO DO : is deviatoric computation more efficient than direct computation ?
+    /// Computes the equivalent stress from a tensor in Voigt notation, using the deviatoric stress
+    double devEquivalentStress(const VoigtTensor2& stressTensor);
+    /// Evaluates the Von Mises yield function for given stress tensor (in Voigt notation), using the deviatoric stress
+    double devVonMisesYield(const VoigtTensor2& stressTensor, const double yieldStress);
+    /// Computes the Von Mises yield function gradient (in Voigt notation) at a given stress tensor (in Voigt notation),
+    ///  using the deviatoric stress
+    auto devVonMisesGradient(const VoigtTensor2& stressTensor) -> VoigtTensor2;
+
+    //Methods called by addForce, addDForce and addKToMatrix when deforming plasticly
+    void accumulateNonLinearForce(VecDeriv& f, const VecCoord& x, int i, Index a, Index b);
+    void applyNonLinearStiffness(VecDeriv& df, const VecDeriv& dx, int i, Index a, Index b, double fact);
+    void updateTangentStiffness(int i, Index a, Index b);
+
+
+    /**********************************************************/
+
+
+    const VecElement *m_indexedElements;
+
+    Data<Real> d_poissonRatio;
+    Data<Real> d_youngModulus;
+    Data<Real> d_initialYieldStress;
+    Data<Real> d_zSection;
+    Data<Real> d_ySection;
+    Data<bool> d_useSymmetricAssembly;
+    Data<bool> d_isTimoshenko;
+
+    Data<std::string> d_sectionShape;
+
+    /// Link to be set to the topology container in the component graph.
+    SingleLink<BeamPlasticFEMForceField<DataTypes>, sofa::core::topology::BaseMeshTopology, BaseLink::FLAG_STOREPATH | BaseLink::FLAG_STRONGLINK> l_topology;
+
+    double m_lastUpdatedStep;
+
+    // Threshold used to compare stress tensor norms to 0. See detailed explanation
+    // at the computation of the threshold in the init() method.
+    Real m_stressComparisonThreshold;
+
+    Quat<SReal>& beamQuat(int i);
+
+    BaseMeshTopology* m_topology;
```

Comment:
You already have a variable pointing to the topology (l_topology). I recommend to use it, and remove this one. Also, m_topology is not initialized to nullptr